contrib/experimental/great_expectations_experimental/expectations/expect_yesterday_count_compared_to_avg_equivalent_days_of_week.py

```diff
@@ -0,0 +1,217 @@
+from datetime import datetime, timedelta
+from typing import Dict, List, Optional
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.core.metric_domain_types import MetricDomainTypes
+from great_expectations.execution_engine import (
+    ExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.execution_engine.execution_engine import MetricFunctionTypes
+from great_expectations.expectations.expectation import ColumnExpectation
+from great_expectations.expectations.metrics import ColumnAggregateMetricProvider
+from great_expectations.expectations.metrics.import_manager import sa
+from great_expectations.expectations.metrics.metric_provider import metric_value
+
+TODAY: datetime = datetime(year=2022, month=8, day=10)
+TODAY_STR: str = datetime.strftime(TODAY, "%Y-%m-%d")
+
+DAYS_AGO = {
+    3: TODAY - timedelta(days=3),
+    7: TODAY - timedelta(days=7),
+    14: TODAY - timedelta(days=14),
+    21: TODAY - timedelta(days=21),
+    28: TODAY - timedelta(days=28),
+}
+
+
+def generate_data_sample(n_appearances: dict):
+    data = []
+    for d, n in n_appearances.items():
+        while n > 0:
+            data.append(d)
+            n -= 1
+    return data
+
+
+class ColumnCountsPerDaysCustom(ColumnAggregateMetricProvider):
+    """Metric that calculate daily counts"""
+
+    metric_name = "column.counts_per_days_custom"
+
+    library_metadata = {"tags": ["query-based"], "contributors": ["@itaise", "@hadasm"]}
+
+    @metric_value(
+        engine=SqlAlchemyExecutionEngine,
+        metric_fn_type=MetricFunctionTypes.AGGREGATE_VALUE,
+        domain_type=MetricDomainTypes.COLUMN,
+    )
+    def _sqlalchemy(
+        cls,
+        execution_engine: SqlAlchemyExecutionEngine,
+        metric_domain_kwargs,
+        metric_value_kwargs,
+        metrics,
+        runtime_configuration,
+    ):
+        (
+            selectable,
+            compute_domain_kwargs,
+            accessor_domain_kwargs,
+        ) = execution_engine.get_compute_domain(
+            metric_domain_kwargs, MetricDomainTypes.COLUMN
+        )
+
+        column_name = accessor_domain_kwargs["column"]
+        column = sa.column(column_name)
+        sqlalchemy_engine = execution_engine.engine
+
+        # get counts for dates
+        query = (
+            sa.select([sa.func.Date(column), sa.func.count()])
+            .group_by(column)
+            .select_from(selectable)
+            .order_by(column.desc())
+            .limit(30)
+        )
+        results = sqlalchemy_engine.execute(query).fetchall()
+        return results
+
+
+class ExpectYesterdayCountComparedToAvgEquivalentDaysOfWeek(ColumnExpectation):
+    """Expect No missing days in date column"""
```

Comment:
Would you mind elaborating here in the docstring and/or expectation name to make it more clear? Specifically that it is calculating relative to the specific day of the week (e.g. Tues, Fri, etc...) and that it is looking back at the last 4 weeks?

Reply:
Of course